src/excmds.ts

```diff
@@ -3072,7 +3072,20 @@ export async function bmark(url?: string, ...titlearr: string[]) {
     dupbmarks.map(bookmark => browser.bookmarks.remove(bookmark.id))
     if (dupbmarks.length != 0) return
     const path = title.substring(0, title.lastIndexOf("/") + 1)
-    // TODO: if title is blank, get it from the page.
+
+
+    // if title is blank, get it from the current page.
+    // technically could race condition if someone switched tabs REALLY quick after
+    // bookmarking, but too unlikely to bother with for now
+    if (title == "") {
```

Comment:
You need to check if `url` is defined as an argument, so move all of this above the ternary `url = url ===` line, otherwise you'll get your current tab's title for unrelated things.

Reply:
I'm not sure I follow you - what kind of call to bmark are you thinking of that would be an 'unrelated thing'?

Reply:
e.g. `bmark google.com` on this page would end up with the title from this page

Reply:
`bmark https://www.google.com` while you were on a random page ends up with the title from that page, not "Google" or "google.com"